src/main/java/org/apache/maven/plugin/compiler/AbstractCompilerMojo.java

```diff
@@ -775,36 +775,16 @@ else if ( CompilerConfiguration.CompilerReuseStrategy.ReuseSame.getStrategy().eq
 
         IncrementalBuildHelperRequest incrementalBuildHelperRequest = null;
 
-        if ( useIncrementalCompilation )
+        if ( !useIncrementalCompilation )
         {
-            getLog().debug( "useIncrementalCompilation enabled" );
+            getLog().debug( "useIncrementalCompilation disabled" );
             try
             {
                 canUpdateTarget = compiler.canUpdateTarget( compilerConfiguration );
 
                 sources = getCompileSources( compiler, compilerConfiguration );
-                
+                compilerConfiguration.setSourceFiles( sources );
                 preparePaths( sources );
-
-                incrementalBuildHelperRequest = new IncrementalBuildHelperRequest().inputFiles( sources );
-
-                // CHECKSTYLE_OFF: LineLength
-                if ( ( compiler.getCompilerOutputStyle().equals( CompilerOutputStyle.ONE_OUTPUT_FILE_FOR_ALL_INPUT_FILES ) && !canUpdateTarget )
```

Comment:
It seems to me that here we are losing this test.

Reply:
this simplification is on purpose for regular behavior and maintainability: we are not using incremental compilation, so we consider all sources.

the else branch is responsible for detecting stale changes.